src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -3436,5 +3455,736 @@ public void Deconstruct(out int a, out int b)
+            var xType = ((FieldSymbol)xSymbol).Type;
+            Assert.True(xType.IsErrorType());
+            Assert.Equal("var", xType.ToTestDisplayString());
+
+            var y = GetDeconstructionVariable(tree, "y");
+            var ySymbol = model.GetDeclaredSymbol(y);
+            Assert.Equal("var Script.y", ySymbol.ToTestDisplayString());
+            var yType = ((FieldSymbol)ySymbol).Type;
+            Assert.True(yType.IsErrorType());
+            Assert.Equal("var", yType.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void FailedCircularInferenceInDeconstructionInScript()
+        {
+            var source =
+@"
+var (x1, x2) = (x2, x1);
+";
+
+            var comp = CreateCompilationWithMscorlib45(source, parseOptions: TestOptions.Script, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            comp.GetDeclarationDiagnostics().Verify(
+                // (2,10): error CS7019: Type of 'x2' cannot be inferred since its initializer directly or indirectly refers to the definition.
+                // var (x1, x2) = (x2, x1);
+                Diagnostic(ErrorCode.ERR_RecursivelyTypedVariable, "x2").WithArguments("x2").WithLocation(2, 10),
+                // (2,6): error CS7019: Type of 'x1' cannot be inferred since its initializer directly or indirectly refers to the definition.
+                // var (x1, x2) = (x2, x1);
+                Diagnostic(ErrorCode.ERR_RecursivelyTypedVariable, "x1").WithArguments("x1").WithLocation(2, 6)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var x1 = GetDeconstructionVariable(tree, "x1");
+            var x1Symbol = model.GetDeclaredSymbol(x1);
+            var x1Ref = GetReference(tree, "x1");
+            Assert.Equal("var Script.x1", x1Symbol.ToTestDisplayString());
+            VerifyModelForDeconstructionField(model, x1, x1Ref);
+            var x1Type = ((FieldSymbol)x1Symbol).Type;
+            Assert.True(x1Type.IsErrorType());
+            Assert.Equal("var", x1Type.Name);
+
+            var x2 = GetDeconstructionVariable(tree, "x2");
+            var x2Symbol = model.GetDeclaredSymbol(x2);
+            var x2Ref = GetReference(tree, "x2");
+            Assert.Equal("var Script.x2", x2Symbol.ToTestDisplayString());
+            VerifyModelForDeconstructionField(model, x2, x2Ref);
+            var x2Type = ((FieldSymbol)x2Symbol).Type;
+            Assert.True(x2Type.IsErrorType());
+            Assert.Equal("var", x2Type.Name);
+        }
+
+        [Fact]
+        public void FailedCircularInferenceInDeconstructionInScript2()
+        {
+            var source =
+@"
+var (x1, x2) = (y1, y2);
+var (y1, y2) = (x1, x2);
+";
+
+            var comp = CreateCompilationWithMscorlib45(source, parseOptions: TestOptions.Script, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            comp.GetDeclarationDiagnostics().Verify(
```

Comment:
That should be fine.

---
In reply to: [80316134](https://github.com/dotnet/roslyn/pull/13957#discussion_r80316134) [](ancestors = 80316134)